src/main/java/org/wildfly/security/http/impl/FormAuthenticationMechanism.java

```diff
@@ -205,6 +206,16 @@ private void attemptAuthentication(HttpServerRequest request) throws HttpAuthent
     }
 
     private boolean authorize(String username, HttpServerRequest request) throws HttpAuthenticationException {
+        if (log.isTraceEnabled()) {
```

Comment:
This block is a little worrying for me, would prefer not to be taking any direct action like obtaining the session scope within an isTraceEbabled block.

When I was in support I had a web services support case that did something similar, it had a side effect that a bug disappeared when debug logging was enabled so really better to try and log what we know rather than trying to obtain information to log.